mono/metadata/object.c

```diff
@@ -7328,30 +7289,23 @@ mono_string_from_utf32_checked (mono_unichar4 *data, MonoError *error)
 }
 
 static char *
-mono_string_to_utf8_internal (MonoMemPool *mp, MonoImage *image, MonoString *s, gboolean ignore_error, MonoError *error)
+mono_string_to_utf8_internal (MonoMemPool *mp, MonoImage *image, MonoStringHandle s, MonoError *error)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
-	char *r;
-	char *mp_s;
-	int len;
-
-	if (ignore_error) {
-		r = mono_string_to_utf8_ignore (s);
-	} else {
-		r = mono_string_to_utf8_checked (s, error);
-		if (!mono_error_ok (error))
-			return NULL;
-	}
+	char * const r = mono_string_handle_to_utf8 (s, error);
+	if (!mono_error_ok (error))
```

Comment:
Could be `is_ok (error)`

Reply:
I think I was preserving what was there, for now

Reply:
Let me clean this this in the entire file later and thoroughly.